lib/promoted/ruby/client/request_builder.rb

```diff
@@ -2,52 +2,49 @@ module Promoted
   module Ruby
     module Client
       class RequestBuilder
-        attr_reader   :session_id, :user_info,
+        attr_reader   :session_id, :only_log, :experiment,
                       :view_id, :insertion, :to_compact_delivery_insertion_func,
                       :request_id, :full_insertion, :use_case, :request, :to_compact_metrics_insertion_func
 
-        def initialize params={}
-          @only_log                = params[:only_log] || false
-        end
+        attr_accessor :timing, :user_info, :platform_id
+
+        def initialize;end
 
         # Populates request parameters from the given arguments, presumed to be a hash of symbols.
         def set_request_params args = {}
           @request                 = args[:request] || {}
+          @experiment              = args[:experiment]
           @session_id              = request[:session_id]
           @platform_id             = request[:platform_id]
-          @cohort_membership       = request[:cohort_membership]
           @view_id                 = request[:view_id]
-          @use_case                = Promoted::Ruby::Client::USE_CASES[request[:use_case]] || 'UNKNOWN_USE_CASE'
+          @use_case                = Promoted::Ruby::Client::USE_CASES[request[:use_case]] || Promoted::Ruby::Client::USE_CASES['UNKNOWN_USE_CASE']
           @full_insertion          = args[:full_insertion]
           @request_id              = SecureRandom.uuid
           @user_info               = request[:user_info] || { :user_id => nil, :log_user_id => nil}
           @timing                  = request[:timing] || { :client_log_timestamp => Time.now.to_i }
-
+          @only_log                = request[:only_log]
           @to_compact_metrics_insertion_func       = args[:to_compact_metrics_insertion_func]
           @to_compact_delivery_insertion_func      = args[:to_compact_delivery_insertion_func]
         end
 
         # Only used in delivery
         def new_cohort_membership_to_log
-          return nil unless request[:experiment]
-          cohort_membership = Hash[request[:experiment]]
-          if !cohort_membership[:platform_id] && request[:platform_id]
-            cohort_membership[:platform_id] = request[:platform_id];
+          return nil unless @experiment
+          if !@experiment[:platform_id] && @platform_id
+            @experiment[:platform_id] = @platform_id
           end
-          if !cohort_membership[:user_info] && request[:user_info]
-            cohort_membership[:user_info] = request[:user_info]
+          if !@experiment[:user_info] && @user_info
```

Comment:
might be more confusing to jump and forth between experiment and cohort_membership.

Reply:
Agree...I also pulled this forward from the TS client, see delivery-request.ts.